modules/suggestions.py

```diff
@@ -0,0 +1,128 @@
+from typing import Union
+
+import discord
+from discord import ui
+from discord.ext import commands
+
+import core
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TypeSelect(ui.Select["TypeView"]):
+    def __init__(
+        self,
+        *,
+        original_author: Union[discord.Member, discord.User],
+        suggestion: str,
+        webhook: discord.Webhook,
+        message: discord.Message | None = None,
+    ):
+        super().__init__()
+        self.original_author = original_author
+        self.suggestion = suggestion
+        self.webhook = webhook
+        self.message = message
+
+        self.add_option(label="Guild", value="1", description="This suggestion applies to the guild.")
+        self.add_option(
+            label="Pythonistabot", value="2", description="This suggestion applies to Pythonistabot.", emoji="\U0001f916"
+        )
+        self.add_option(
+            label="TwitchIO",
+            value="3",
+            description="This suggestion applies to TwitchIO.",
+            emoji="<:twitchio_illegal:1014831894417915984>",
+        )
+        self.add_option(
+            label="Wavelink",
+            value="4",
+            description="This suggestion applies to Wavelink.",
+            emoji="<:wavelink:753851443093438545>",
+        )
+
+    async def callback(self, interaction: discord.Interaction):
+        if self.original_author != interaction.user:
+            return await interaction.response.send_message(
+                f"This menu is not for you. See `{core.CONFIG['prefix']}suggest` to make a suggestion!", ephemeral=True
+            )
+        author = self.original_author
+        if self.values[0] == "1":  # Guild Suggestion
+            suggestion_type = "the guild"
+        elif self.values[0] == "2":  # Pythonistabot Suggestion
+            suggestion_type = "Pythonistabot"
+        elif self.values[0] == "3":
+            suggestion_type = "TwitchIO"
+        else:
+            suggestion_type = "Wavelink"
+        await interaction.response.send_message("Your suggestion has been sent!")
+        embed = discord.Embed(
+            title=f"Suggestion for {suggestion_type}", description=self.suggestion, timestamp=datetime.now(), color=0x7289DA
+        )
+        embed.set_author(name=author.name, icon_url=author.display_avatar)
+        embed.set_footer(text=f"Suggestion sent by {author} (ID: {author.id})")
+        await self.webhook.send(embed=embed, avatar_url=author.display_avatar, username=author.name)
+        if self.message:
+            await self.message.delete()
+
+
+class TypeView(ui.View):
+    def __init__(self, *, original_author: Union[discord.Member, discord.User], suggestion: str, webhook: discord.Webhook):
+        super().__init__(timeout=180)
+        self.original_author = original_author
+        self.add_item(TypeSelect(original_author=original_author, suggestion=suggestion, webhook=webhook))
+        self.message: discord.Message
```

Comment:
The more accurate thing to do here is to move `message` as a class level type annotation and assign there. That way we don't need it as a parameter since we technically duck-type it in.

```suggestion
class TypeView(ui.View):
    message: discord.Message | discord.WebhookMessage
```